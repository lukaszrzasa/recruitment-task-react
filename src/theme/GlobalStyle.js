import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  body, html {
    margin: 0;
    padding: 0;
    overflow: hidden;
  }
  * {
    overflow-wrap: break-word;
  }
`;

export default GlobalStyle;