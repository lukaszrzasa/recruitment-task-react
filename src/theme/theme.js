const colors = {
  violet: '#b33ed6',
  black: '#222',
  default: '#222',
  gray: '#809396',
  gold: '#ffdf13',
  white: '#fff',
  red: '#be2d10',
  green: '#13ad25',
  transparent: 'transparent',
};

const sizes = {
  xxl:200,
  xl: 60,
  lg:40,
  md:22,
  sm:18,
  xs: 14,
};

const theme = {
  colors,
  sizes,
};
export default theme;