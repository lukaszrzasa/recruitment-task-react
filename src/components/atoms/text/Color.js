import styled, {css} from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../../theme/theme';
import {getColor} from '../helpers/getColor';


const Color = styled.span`
  color: ${({variant})=>getColor(variant)};
  ${({isBlock}) => isBlock && css`
    display: block;
  `}
`;

Color.propTypes = {
  variant: PropTypes.oneOf(Object.keys(theme.colors)).isRequired,
  isBlock: PropTypes.bool,
};

Color.defaultProps = {
  variant: 'default',
  isBlock: false,
};

export default Color;