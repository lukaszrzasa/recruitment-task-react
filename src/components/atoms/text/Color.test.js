import React from 'react';
import {render} from '@testing-library/react';
import Color from './Color';
import theme from '../../../theme/theme';

const { colors } = theme;

describe('atoms/text/Color component', ()=>{

  it('render properly',()=>{
    const { container } = render(<Color variant="violet" />);
    expect(container.firstChild).toMatchSnapshot()
  });

  it('has good color',()=>{
    const text = 'Text that\'ll be tested';
    const { getByText, rerender } = render(<Color variant="violet">{text}</Color>);
    expect( getByText(text) ).toHaveStyle(`color: ${colors.violet}`);
    rerender(<Color variant="black">{text}</Color>);
    expect( getByText(text) ).toHaveStyle(`color: ${colors.black}`);
  });

  it('renders as block',()=>{
    const text = 'Text that\'ll be tested';
    const { getByText } = render(<Color variant="violet" isBlock={true}>{text}</Color>);
    expect( getByText(text) ).toHaveStyle(`display: block`);
  });

});