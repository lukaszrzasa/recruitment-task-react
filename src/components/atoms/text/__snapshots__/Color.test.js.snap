// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/text/Color component render properly 1`] = `
<span
  class="sc-AxjAm bojodx"
/>
`;
